src/transaction.js

```diff
@@ -441,15 +438,10 @@ Transaction.prototype.toBuffer = function (buffer, initialOffset) {
     writeVarSlice(txOut.script)
   })
 
-  if (serializeWitnesses) {
-    var isNull = true
+  if (hasWitnesses) {
     this.ins.forEach(function (input) {
       writeVector(input.witness)
-      isNull = isNull && input.witness.length === 0
     })
-    if (isNull) {
-      throw new Error('Transaction has superfluous witness data')
-    }
```

Comment:
We can't even test this check yet because there's no way to ask for only the base serialization.